### DISCUSSION TITLE
Is there a way to achieve this today somehow?

### DISCUSSION THREAD

**Discussion by frederikhors** [ent/ent#2631] [Q&A] [answered]
Hi guys!

We're in the middle of transaction from Bun (like Gorm) to Ent.

What a joy working with Ent! Thank you!

### Problem

With Bun (and others before) we were using something like the below for entities like "tags/notes":

```go
type Tag struct {
  ID            uint64
  TrackableType *string
  TrackableID   *uint64
  Label         string `bun:",unique,notnull"`
}

type Player struct {
  ID   uint64
  Name string
  Tags []*Tag `bun:"rel:has-many,join:id=trackable_id,join:type=trackable_type,polymorphic"`
}
```

So basically I can add a `[]Tag` on an entity without declaring in `Tag` which entities should have it.

This is what they call "Polymorphism" (at least I think :smile:).

[I know you were talking about it already](https://github.com/ent/ent/issues/1048), but my question is:

**is there a way to do this TODAY somehow?**

Because this can be a really blocking issue for us (we have tons of these poly structs).

_Thank you again for your commitment._

**Comment by a8m** [comment]
Hey @frederikhors!

> So basically I can add a []Tag on an entity without declaring in Tag which entities should have it.

Is it a one-2-many or many-2-many relationship? If it's one-2-many, you can achieve this today by defining tags on users and without defining the back-reference from a tag to its user, and if you want to save the repeated declaration you can use a mixed-in schema for this. If it's many-2-many, I'd love if you can elaborate on the data model because from looking at the `Tag` object, it looks like it can have only one  `TrackableID`.

**Comment by frederikhors** [comment, marked as answer]
Polymorphism is tracked here: https://github.com/ent/ent/issues/1048.
